File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    answer = -1
    sameCount = 0
    while sameCount != len(arr):
        sameCount = 0
        prev = arr[:]
        for i in range(len(arr)):
            if arr[i] >= 50 and arr[i] % 2 == 0:
                arr[i] = arr[i] // 2
            elif arr[i] < 50 and arr[i] % 2 ==1:
                arr[i] = arr[i] * 2 + 1
            if arr[i] == prev[i]:
                sameCount+=1
        answer +=1
    return answer

File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
class Solution {
    public int solution(int[] arr) {
        int answer = -1;
        int sameCount = 0;
        while(sameCount != arr.length){
            sameCount = 0;
            int[] prev = new int[arr.length];
            for(int i =0;i<prev.length;i++){
                prev[i] = arr[i];
            }
            for(int i=0;i<arr.length;i++){
                if(arr[i] >= 50 && arr[i]%2 == 0){
                    arr[i] = arr[i] / 2;
                } else if(arr[i] < 50 && arr[i] %2 == 1){
                    arr[i] = arr[i] * 2 + 1;
                }
                if(arr[i] == prev[i])
                    sameCount++;
            }
            answer++;
        }
        return answer;
    }
}